extensions/interactions/base.py

```diff
@@ -263,3 +263,18 @@ def get_rule_param_type(self, rule_name, rule_param_name):
                 return param_type
         raise Exception(
             'Rule %s has no param called %s' % (rule_name, rule_param_name))
+
+    def get_hint_proto_list(self, hints):
+        """Gets the list of HintDto proto for given hints domain object list.
+
+        Args:
+            hints: list(Hint). List of hints.
+
+        Returns:
+            list. The list of HintDto proto object.
+        """
+        hints_proto = {}
+        if hints is not None:
+            hints_proto = [hint.to_android_hint_proto() for hint in hints]
```

Comment:
Why are you changing a dict (the default value) into a list?

Reply:
This function returns the list of hint proto which is required to set to proto directly. 
```proto
repeated HintDto hints = 4;
```